run-tests.py

```diff
@@ -26,9 +26,9 @@ def run_sbt_tests(root_dir, scala_version=None):
     print("##### Running SBT tests #####")
     sbt_path = path.join(root_dir, path.join("build", "sbt"))
     if scala_version is None:
-        run_cmd([sbt_path, "clean", "+test"], stream_output=True)
+        run_cmd([sbt_path, "clean", "+test", "coverageAggregate"], stream_output=True)
```

Comment:
Note: We can make the `coverageAggregate` a separate step in the CI pipeline after "Run Scala/Java and Python tests".

Reply:
without understanding the impact of enabling code coverage on test runtime (which is already pretty long), I dont think we should enable this automatically. Can you make it such it will enabled only when run-tests.py is executed in some param or env variable?

Reply:
Regarding test runtimes, here's a summary of how long the "Run Scala/Java and Python tests" step took for this branch vs master branch:

- 2.12.14:
    - delta-io:master: [1h 37m 5s](https://github.com/delta-io/delta/runs/7617834803?check_suite_focus=true)
    - hedibejaoui:feature-add-sbt-test-coverage-report: [1h 39m 57s](https://github.com/delta-io/delta/runs/7677044440?check_suite_focus=true)
- 2.13.5:
    - delta-io:master: [1h 17m 27s](https://github.com/delta-io/delta/runs/7617834917?check_suite_focus=true)
    - hedibejaoui:feature-add-sbt-test-coverage-report: [1h 24m 45s](https://github.com/delta-io/delta/runs/7677044546?check_suite_focus=true)

I'll make changes such as run-tests.py will execute with coverage disabled by default, and enable it during the CI build via a parameter or environment variable like you suggested.